### PR TITLE
Bump BDK to 0.21

### PR DESCRIPTION
**Pull request by klochowicz**
it also updates maia and removes our fork of electrum-client

### CODE REVIEW

**Review by holzeis — APPROVED** [review]
LGTM 👍 

Will this also fix the warnings in the logs?

```
WARN Send message every interval{interval_secs=180}:xtra_actor_request{actor_type=daemon::wallet::Actor<bdk::blockchain::electrum::ElectrumBlockchain, sled::tree::Tree> message_type=daemon::wallet::Sync}:xtra_message_handler:Handle Sync:Sync wallet:Sync wallet database with blockchain: electrum_client::raw_client: got error for req_id 761: JSON(Error("EOF while parsing a value", line: 1, column: 0))    
```

**Comment by klochowicz** [comment]
> LGTM 👍
> 
> Will this also fix the warnings in the logs?
> 
> ```
> WARN Send message every interval{interval_secs=180}:xtra_actor_request{actor_type=daemon::wallet::Actor<bdk::blockchain::electrum::ElectrumBlockchain, sled::tree::Tree> message_type=daemon::wallet::Sync}:xtra_message_handler:Handle Sync:Sync wallet:Sync wallet database with blockchain: electrum_client::raw_client: got error for req_id 761: JSON(Error("EOF while parsing a value", line: 1, column: 0))    
> ```

this is indeed the hope! after merging this, the idea is to closely monitor testnet.

File: Cargo.toml [hunk 1]
```diff
@@ -26,8 +26,10 @@ resolver = "2"
 
 [patch.crates-io]
 xtra = { git = "https://github.com/Restioson/xtra", rev = "285b3e986013888cb68b9219464ef325d2468c2c" } # Unreleased
-maia = { git = "https://github.com/comit-network/maia", rev = "fc6b78b98407b10b55f8cfd152062ad77f98cd9f" } # Unreleased
-maia-core = { git = "https://github.com/comit-network/maia", tag = "0.1.1", package = "maia-core" } # Pinned to support maia 0.1 and 0.2
+maia = { git = "https://github.com/comit-network/maia", rev = "9899c9eda1f7685493aecdd7f8ba9124787056bd" }
+maia-core = { git = "https://github.com/comit-network/maia", rev = "9899c9eda1f7685493aecdd7f8ba9124787056bd", package = "maia-core" }
+
+
```
Comment (luckysori) [inline Cargo.toml]: 🔧 Unneeded newlines.

File: daemon/src/wallet.rs [hunk 2]
```diff
@@ -167,8 +167,9 @@ where
                 .context("Failed to sync wallet")
         })?;
 
-        let balance =
-            tracing::debug_span!("Get wallet balance").in_scope(|| self.wallet.get_balance())?;
+        let balance = tracing::debug_span!("Get wallet balance")
+            .in_scope(|| self.wallet.get_balance())?
+            .get_spendable();
```
Comment (luckysori) [inline daemon/src/wallet.rs:172]: 🙃 Should we do this in a separate commit? It might be easier to find this change if it's not included in the upgrade commit.

File: Cargo.toml [hunk 3]
```diff
@@ -28,10 +28,7 @@ resolver = "2"
 xtra = { git = "https://github.com/Restioson/xtra", rev = "285b3e986013888cb68b9219464ef325d2468c2c" } # Unreleased
 maia = { git = "https://github.com/comit-network/maia", rev = "9899c9eda1f7685493aecdd7f8ba9124787056bd" }
 maia-core = { git = "https://github.com/comit-network/maia", rev = "9899c9eda1f7685493aecdd7f8ba9124787056bd", package = "maia-core" }
-
-
```
Comment (luckysori) [inline Cargo.toml]: Ah, they disappear in the last commit.

**Review by luckysori — APPROVED** [review]

File: daemon/src/wallet.rs (hunk 2 above)
Comment (klochowicz) [reply daemon/src/wallet.rs:172]: I was considering it, but it doesn't compile without it (breaking `git bisect` etc). Maybe I can be clearer in the commit description that BDK changed balance API in 0.21 and now you need to explicitly specify which one you want?

Comment (luckysori) [reply daemon/src/wallet.rs:172]: That makes sense! Adding a bit of context can't hurt :D

**Comment by klochowicz** [comment]
@luckysori I have removed the last commit (removing our fork of `electrum-rust-client`, as it turns out that only bdk 0.22 will use the newer electrum-rust-client, which will not require us manually patching that dependency.
